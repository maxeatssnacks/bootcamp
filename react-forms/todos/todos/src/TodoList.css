.TodoApp {
    max-width: 600px;
    margin: 0 auto;
    padding: 20px;
    font-family: Arial, sans-serif;
}

.TodoApp h3 {
    font-size: 32px;
    text-align: center;
    color: #b7b26b;
}

.TodoList {
    list-style-type: none;
    padding: 0;
}