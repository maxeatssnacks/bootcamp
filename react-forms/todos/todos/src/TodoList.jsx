import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import { v4 as uuid } from 'uuid';
import "./TodoList.css";

const TodoList = () => {
    const [tasks, setTasks] = useState([]);

    const addTodo = (newTodo) => {
        setTasks(tasks => [...tasks, { ...newTodo, id: uuid() }]);
    };

    const removeTodo = (id) => {
        setTasks(tasks => tasks.filter(task => task.id !== id));
    };

    return (
        <div className="TodoApp">
            <h3>Let's Git R Done</h3>
            <NewTodoForm addTodo={addTodo} />
            <ul className="TodoList">
                {tasks.map(({ id, task }) => (
                    <Todo key={id} id={id} task={task} removeTodo={removeTodo} />
                ))}
            </ul>
        </div>
    );
};

export default TodoList;