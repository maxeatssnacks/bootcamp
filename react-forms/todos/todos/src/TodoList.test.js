import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

// Smoke test
test('TodoList renders without crashing', () => {
    render(<TodoList />);
});

// Snapshot test
test('TodoList matches snapshot', () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

// Critical business logic test
test('can add a new todo', () => {
    render(<TodoList />);

    // Find the input and add button
    const input = screen.getByPlaceholderText('Enter a task here');
    const addButton = screen.getByText('Add Task');

    // Add a new todo
    fireEvent.change(input, { target: { value: 'New Todo' } });
    fireEvent.click(addButton);

    // Check if the new todo is in the document
    expect(screen.getByText('New Todo')).toBeInTheDocument();
});

test('can remove a todo', () => {
    render(<TodoList />);

    // Add a new todo
    const input = screen.getByPlaceholderText('Enter a task here');
    const addButton = screen.getByText('Add Task');
    fireEvent.change(input, { target: { value: 'Todo to remove' } });
    fireEvent.click(addButton);

    // Find and click the delete button
    const deleteButton = screen.getByText('Delete');
    fireEvent.click(deleteButton);

    // Check if the todo is removed from the document
    expect(screen.queryByText('Todo to remove')).not.toBeInTheDocument();
});