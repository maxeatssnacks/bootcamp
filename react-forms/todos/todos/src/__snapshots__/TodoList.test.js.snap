// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoApp"
  >
    <h3>
      Let's Git R Done
    </h3>
    <div
      class="NewTodoForm"
    >
      <form>
        <input
          id="task"
          name="task"
          placeholder="Enter a task here"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Add Task
        </button>
      </form>
    </div>
    <ul
      class="TodoList"
    />
  </div>
</DocumentFragment>
`;
